File: Source Code/BatchExecutor.cpp
/*
 * Title: Batch Executor
 * Language: C++
 * Version: 1.0
 */
#include <iostream>
#include <Windows.h>
#include <locale>
#include <io.h>
using namespace std;

int main()
{
	char fullpath[MAX_PATH] {};
    GetModuleFileNameA(nullptr, fullpath, MAX_PATH);

	char convert_ext[4] = "BAT";

	int fullpath_index = sizeof(fullpath) / sizeof(char);
	int last_index = 0;
	int last_period_index = 0;
	int last_backslash_index = 0;

	for (int i = MAX_PATH; i > 0; i--) {
		if (fullpath[i - 1] != NULL && last_index == 0) {
			last_index = i;
		}
		else if (fullpath[i - 1] == '.' && last_period_index == 0) {
			last_period_index = i;
		}
		else if (fullpath[i - 1] == '\\' && last_backslash_index == 0) {
			last_backslash_index = i;
		}
		if (last_index != 0 && last_period_index != 0 && last_backslash_index != 0) {
			break;
		}
	}
	for (int i = 0; i < sizeof(convert_ext) / sizeof(char) - 1; i++) {
		fullpath[last_period_index + i] = convert_ext[i];
	}

	//파일이 있다면 IF FILE EXISTS
	if (_access(fullpath, 0) == 0) {
		last_index += 3;
		char* batFullpath = new char[last_index] {'\"', };
		strcat_s(batFullpath, last_index, fullpath);
		strcat_s(batFullpath, last_index, "\"");

		system(batFullpath); //실행부 RUN BATCH FILE
		delete[] batFullpath;
	}
	//에러메세지 ERROR MESSAGE
	else {
		char* path = new char[last_backslash_index] {};
		char* filename = new char[last_index - last_backslash_index + 1] {};

		for (int i = 0; i < last_index; i++) {
			if (i < last_backslash_index - 1) {
				path[i] = fullpath[i];
			}
			else if (i > last_backslash_index - 1) {
				filename[i - last_backslash_index] = fullpath[i];
			}
		}

		LANGID winId = GetUserDefaultUILanguage();
		//한글
		if ((winId & 0xff) == LANG_KOREAN) {
			cout << "\"" << path << "\"에 \"" << filename << "\"이 없습니다." << endl;
			system("PAUSE");
			delete[] path;
			delete[] filename;
			exit(0);
		}
		//English
		cout << "There is no \"" << filename << "\" in \"" << path << "\"." << endl;
		system("PAUSE");
		delete[] path;
		delete[] filename;
	}
}

File: Source Code/BatchExecutor.cs
/*
 * Title: Batch Executor
 * Language: C#
 * Version: 1.0
 */
using System;
using System.Diagnostics;
using System.Globalization;
using System.IO;
using System.Text.RegularExpressions;

namespace BatchExecutor
{
    class BatchExecutor
    {
        static void Main(string[] args)
        {
            string batFullpath = Regex.Replace(System.Reflection.Assembly.GetExecutingAssembly().Location, @"(?<=\.)exe$", "BAT");

            //파일이 있다면 IF FILE EXISTS
            if (File.Exists(batFullpath))
            {
                Process.Start(batFullpath); //실행부 RUN BATCH FILE
            }
            else //에러메세지 ERROR MESSAGE
            {
                string path = Path.GetDirectoryName(batFullpath);
                string name = Path.GetFileName(batFullpath);

                //한글
                if (CultureInfo.CurrentCulture.Name.Split('-')[0].Equals("ko"))
                {
                    Console.WriteLine("\"" + path + "\"에 \"" + name + "\"이 없습니다.");
                    Console.WriteLine("계속하려면 아무 키나 누르십시오 . . .");
                    Console.ReadKey(); // PAUSE
                    Environment.Exit(0);
                }
                //English
                Console.WriteLine("There is no \"" + name + "\" in " + "\"" + path + "\".");
                Console.WriteLine("Press any key to continue . . .");
                Console.ReadKey(); // PAUSE
            }
        }
    }
}
